x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -135,10 +135,23 @@ public DataFrameTransformState getState() {
 
     void initializePreviousStats(DataFrameIndexerTransformStats stats) {
         previousStats.merge(stats);
+        if (stats.getCurrentRunStartTime() != null) {
+            previousStats.setCurrentRunStartTime(stats.getCurrentRunStartTime().getTime());
+        }
     }
 
     public DataFrameIndexerTransformStats getStats() {
-        return new DataFrameIndexerTransformStats(previousStats).merge(indexer.getStats());
+        return mergePreviousStats(indexer.getStats());
+    }
+
+    private DataFrameIndexerTransformStats mergePreviousStats(final DataFrameIndexerTransformStats stats) {
+        DataFrameIndexerTransformStats tempStats = new DataFrameIndexerTransformStats(stats).merge(previousStats);
+        // Handle the case for when task is migrated between nodes, but a new run did not start, so we should keep
+        // the previous stats run start time.
+        if (tempStats.getCurrentRunStartTime() == null && previousStats.getCurrentRunStartTime() != null) {
+            tempStats.setCurrentRunStartTime(previousStats.getCurrentRunStartTime().getTime());
```

Comment:
This is a tricky situation this is the best way I could see how to handle it currently. 

There is the chance that the node dies in the middle of an indexing run and the task is migrated. So, we still have the previous progression statistics and start time persisted in the previous stats. Additionally, since it is not an "initial run" after the task is migrated to a new node, the progression information will not be present in the object. So, the best way I could think of is simply merging and then checking if the current stats do not have `currentRunStartTime` defined yet, and then setting it to what was previous stored (if anything was).